Question 1 : What is the total number and percentage of missing data in 2014 to 3 decimal places?

In [11]:
import pandas as pd

# Load the dataset from the extracted CSV file
csv_path = 'FoodBalanceSheets_E_Africa_NOFLAG.csv'
data = pd.read_csv(csv_path, encoding='latin-1')

# Calculate the number of missing values in the column for the year 2014
missing_2014 = data['Y2014'].isnull().sum()

# Calculate the total number of entries in the dataset
total_entries_2014 = len(data['Y2014'])

# Calculate the percentage of missing data
percentage_missing_2014 = (missing_2014 / total_entries_2014) * 100

# Format the output to 3 decimal places
missing_2014, total_entries_2014, formatted_percentage_2014 = missing_2014, total_entries_2014, f"{percentage_missing_2014:.3f}%"

missing_2014, formatted_percentage_2014


(1589, '2.607%')

Question 4 : which year had the least correlation with 'Element Code'

In [ ]:
# Load the dataset from the CSV file again to ensure we're working with the correct data
data = pd.read_csv(csv_path, encoding='latin-1')

# Calculate the correlation coefficients for 'Element Code' with each specified year
correlation_2014 = data['Element Code'].corr(data['Y2014'])
correlation_2015 = data['Element Code'].corr(data['Y2015'])
correlation_2016 = data['Element Code'].corr(data['Y2016'])
correlation_2017 = data['Element Code'].corr(data['Y2017'])

# Determine the year with the least correlation
year_correlations = {
    '2014': abs(correlation_2014),
    '2015': abs(correlation_2015),
    '2016': abs(correlation_2016),
    '2017': abs(correlation_2017)
}

year_correlations


Based on the absolute values of the correlation coefficients calculated between 'Element Code' and each of the years:

    2014: 0.0245
    2015: 0.0239
    2016: 0.0234
    2017: 0.0243

The year 2016 has the least correlation with 'Element Code', with a correlation coefficient closest to 0.

Question 5 : which area had the highest sum in 2017

In [3]:
# Perform a groupby operation on 'Area' and then sum up the values for 2017
area_grouped = data.groupby('Area')['Y2017'].sum()

# Find the area with the highest sum in 2017
highest_area_2017 = area_grouped.idxmax()
highest_sum_2017 = area_grouped.max()

highest_area_2017, highest_sum_2017


('Nigeria', 1483268.23)

Question 6 : which year has the highest sum of 'Stock Variation' 

In [4]:
# First, filter the data for the 'Stock Variation' element
stock_variation_data = data[data['Element'] == 'Stock Variation']

# Now we will sum the values of 'Stock Variation' for each of the years in question
stock_variation_sums = {
    '2014': stock_variation_data['Y2014'].sum(),
    '2015': stock_variation_data['Y2015'].sum(),
    '2016': stock_variation_data['Y2016'].sum(),
    '2017': stock_variation_data['Y2017'].sum()
}

# Determine which year had the highest sum of 'Stock Variation'
highest_stock_variation_year = max(stock_variation_sums, key=stock_variation_sums.get)
highest_stock_variation_year, stock_variation_sums[highest_stock_variation_year]


('2014', 58749.83)

The year with the highest sum of 'Stock Variation' is 2014. 

Question 8 : 

In [5]:
# Calculate the mean and standard deviation for the year 2017
mean_2017 = data['Y2017'].mean()
std_dev_2017 = data['Y2017'].std()

# Round the results to 2 decimal places
mean_2017_rounded = round(mean_2017, 2)
std_dev_2017_rounded = round(std_dev_2017, 2)

mean_2017_rounded, std_dev_2017_rounded


(140.92, 1671.86)

The mean and standard deviation across the whole dataset for the year 2017 to two decimal places are 140.92 and 1671.86, respectively. 

Question 9 :

In [6]:
# Perform a groupby operation on 'Area' and then sum up the values for 2017
area_sum_2017 = data.groupby('Area')['Y2017'].sum()

# Sort the sums in ascending order and find the 7th lowest sum
sorted_area_sum_2017 = area_sum_2017.sort_values()
seventh_lowest_area = sorted_area_sum_2017.iloc[6]

# Get the name of the area with the 7th lowest sum
seventh_lowest_area_name = sorted_area_sum_2017.index[6]
seventh_lowest_area_name, seventh_lowest_area


('Guinea-Bissau', 19102.77)

Question 12 : find the total sum of Wine produced in 2015 and 2018,

In [7]:
# Perform a groupby operation on 'Item' and then sum up the values for 2015 and 2018
wine_production = data[data['Item'] == 'Wine'].groupby('Item').agg({'Y2015': 'sum', 'Y2018': 'sum'})

# Extract the total production of wine for 2015 and 2018
wine_production_2015_2018 = wine_production.loc['Wine']
wine_production_2015_2018_values = (wine_production_2015_2018['Y2015'], wine_production_2015_2018['Y2018'])
wine_production_2015_2018_values


(4251.81, 4039.32)

The total sum of Wine produced in 2015 was 4251.81, and in 2018 it was 4039.32

Question 14 : 

In [8]:
# Calculate the total number of unique countries (Areas) in the dataset
total_unique_countries = data['Area'].nunique()
total_unique_countries


49

The total number of unique countries in the dataset is 49

Question 16: find the total sum of Processing for the year 2017

In [9]:
# Perform a groupby operation on 'Element' and sum up the values for 2017
processing_sum_2017 = data[data['Element'] == 'Processing'].groupby('Element')['Y2017'].sum()

# Retrieve the total number for 'Processing' in 2017
processing_sum_2017_value = processing_sum_2017.iloc[0]
processing_sum_2017_value


292836.0

The total sum of Processing in 2017 is 292836.00

Question 17: find the total Protein supply quantity in Madagascar for the year 2015

In [10]:
# Filter the dataset for Madagascar and the Protein supply quantity element
madagascar_protein_2015 = data[(data['Area'] == 'Madagascar') & 
                               (data['Element'] == 'Protein supply quantity (g/capita/day)')]

# Sum the values for the year 2015
total_protein_2015 = madagascar_protein_2015['Y2015'].sum()
total_protein_2015


173.04999999999998

The total Protein supply quantity in Madagascar for the year 2015 is 173.05 (rounded to two decimal places).